app/src/atoms/AtomButton.re

```diff
@@ -1,87 +1,100 @@
+      backgroundColor(Color.white()),
+      borderRadius(px(4)),
+      borderStyle(solid),
+      borderWidth(px(1)),
+      borderColor(Color.lightGray()),
+      color(Color.darkGray()),
+      cursor(`pointer),
+      height(px(48)),
+      outlineStyle(`none),
+      padding4(~top=px(13), ~right=px(24), ~bottom=px(14), ~left=px(24)),
+      hover([backgroundColor(Color.almostWhite(~alpha=0.85, ()))]),
+      active([backgroundColor(Color.almostWhite(~alpha=0.2, ()))]),
+    ])
+    << style(Font.title);
 
   let primaryButton =
-    merge([
-      button,
-      style([
-        borderStyle(none),
-        color(Color.white()),
-        backgroundColor(Color.purple()),
-        hover([backgroundColor(Color.purple(~alpha=0.85, ()))]),
-        active([backgroundColor(Color.purple(~alpha=0.75, ()))]),
-      ]),
-    ]);
+    button
+    << style([
+         borderStyle(none),
+         color(Color.white()),
+         backgroundColor(Color.purple()),
+         hover([backgroundColor(Color.purple(~alpha=0.85, ()))]),
+         active([backgroundColor(Color.purple(~alpha=0.75, ()))]),
+       ]);
 
   let secondaryButton =
-    merge([
-      button,
-      style([
-        borderStyle(none),
-        color(Color.white()),
-        backgroundColor(Color.pink()),
-        hover([backgroundColor(Color.pink(~alpha=0.85, ()))]),
-        active([backgroundColor(Color.pink(~alpha=0.75, ()))]),
-      ]),
-    ]);
+    button
+    << style([
+         borderStyle(none),
+         color(Color.white()),
+         backgroundColor(Color.pink()),
+         hover([backgroundColor(Color.pink(~alpha=0.85, ()))]),
+         active([backgroundColor(Color.pink(~alpha=0.75, ()))]),
+       ]);
+
+  let cancelButton =
+    button
+    << style([
+         borderStyle(none),
+         color(Color.gray()),
+         hover([
+           backgroundColor(Color.white()),
+           color(Color.gray(~alpha=0.75, ())),
+         ]),
+         active([
+           backgroundColor(Color.white()),
+           color(Color.gray(~alpha=0.55, ())),
+         ]),
+       ]);
 
   let disabled =
-    merge([
-      button,
-      style([
-        backgroundColor(Color.gray()),
-        borderStyle(none),
-        hover([backgroundColor(Color.gray())]),
-        active([backgroundColor(Color.gray())]),
-      ]),
-    ]);
+    button
+    << style([
+         backgroundColor(Color.gray()),
+         borderStyle(none),
+         hover([backgroundColor(Color.gray())]),
+         active([backgroundColor(Color.gray())]),
+       ]);
+};
+
+module Template = {
+  [@react.component]
+  let make = (~children, ~disabled, ~onClick, ~style) =>
+    <button onClick className={disabled ? Styles.disabled : style} disabled>
+      children
+    </button>;
 };
 
 [@react.component]
-let make = (~children, ~disabled=false) =>
-  <button className={disabled ? Styles.disabled : Styles.button} disabled>
-    children
-  </button>;
+let make = (~children, ~disabled=false, ~onClick=_ => ()) =>
```

Comment:
Would be cool if we can type our onClick props according to the [docs](https://github.com/reasonml/reason-react/blob/521295715e21a510442f8174e3cea2323f3ec80b/src/ReactEvent.rei#L4-L11).

Reply:
Ah, thanks for reminding me, I remember we talked about this. On it.

Reply:
As we just discussed this IRL, the Reason type system is clever enough to infer the types for us and adding the type annotations by hand just makes things more verbose without added value.